File: Fontend/TNT.N8.Web/TNT.N8.Web/ClientApp/src/app/product/components/product-options/model/options.ts
export class OptionsEntityModel {
    id: string;
    categoryId: string;
    name: string;
    description: string;
    note: string;
    enableNumber: boolean | null;
    isShowQuantity: boolean | null;
    createdById: string | null;
    createdDate: string | null;
    updatedById: string | null;
    updatedDate: string | null;
    tenantId: string | null;
    price: number | null;
    productId: string;
    parentId: null | string;
    vat: number;
}
export class SearchOptionTreeResult {
    listOptions: ListOptions[];
    statusCode: number;
    messageCode: null;
}

export class ListOptions {
    categoryName: string;
    optionName: string;
    price: number;
    optionId: string;
    parentId: null | string;
    hasChild: boolean;
    vat: number;
    listOptions: ListOptions[];
}
export class CreateOrUpdateOptionsResult {
    statusCode: number;
    messageCode: string;
    optionsEntityModel: OptionsEntityModel;
}
export class OptionsEntityTable {
    statusCode: number;
    messageCode: string;
    optionsEntityModel: OptionsEntityModel;
}
export class MilestoneConfigurationEntityModel {
    id: string;
    scopeReport: string;
    name: string;
    clientView: boolean | null;
    deadline: string | null;
    status: string;
    content: string;
    employeeId: string | null;
    workingOrder: number | null;
    note: string;
    createdById: string;
    createdDate: string;
    updatedById: string;
    updatedDate: string;
    tenantId: string;
    optionId: string;
}
export class AttributeConfigurationModel {
    id: string;
    dataType: number;
    categoryId: string;
    createdById: string;
    createdDate: string;
    updatedById: string;
    updatedDate: string;
    tenantId: string;
    objectType: number;
    objectId: string;
}
export class CreateOrUpdateOptionsModel {
    optionsEntityModel: OptionsEntityModel;
    attributeConfigurationModel: AttributeConfigurationModel;
    milestoneConfigurationEntityModel: MilestoneConfigurationEntityModel

}

File: Backend/TN.TNM.DataAccess/Databases/Entities/Options.cs
using System;
using System.Collections.Generic;

namespace TN.TNM.DataAccess.Databases.Entities
{
    public partial class Options
    {
        public Guid Id { get; set; }
        public Guid CategoryId { get; set; }
        public Guid? CategoryUnitId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Note { get; set; }
        public Guid CreatedById { get; set; }
        public DateTime CreatedDate { get; set; }
        public Guid UpdatedById { get; set; }
        public DateTime UpdatedDate { get; set; }
        public Guid TenantId { get; set; }
        public decimal? Price { get; set; }
        public Guid? ParentId { get; set; }
        public decimal? Vat { get; set; }
    }
}
